### DISCUSSION TITLE
Help on "OnMove etc."

### DISCUSSION THREAD

**Discussion by Fpaciello1** [sceneview/sceneview#325] [Q&A]
Hi I'm new to scene view / Ar Core, I added some 3D model using the listener onTapAr :

`            onTapAr = { hitResult, _ ->
                if (toolActive) {
                    val anchor = hitResult.createAnchor()
                    lifecycleScope.launch {
                        // Load a new model instance for each tap
                        val modelInstance = GLBLoader.loadModelInstance(
                            context = requireContext(),
                            glbFileLocation = "models/sphere.glb"
                        ) as ModelInstance

                        val sphereModel = ArModelNode(sceneView.engine)
                        
                        sphereModel.scale = Scale(0.03f, 0.03f, 0.03f)
                        for (material in modelInstance.materialInstances) {
                            material.setBaseColor(Color(255f, 0f, 0f, 255f))
                        }
                        sphereModel.modelInstance = modelInstance
                        sphereModel.anchor = anchor
                        // Add the anchor node to the AR scene
                        sceneView.addChild(sphereModel)
                        }
               }`
               
               
Now I would like to add a listener to print some debug messages, like position of the sphereModel and rotation when I move it on the scene. I think I should use this : 
`public open fun onMove(
    detector: MoveGestureDetector,
    e: NodeMotionEvent
): Unit`

But I can't find a way to work with it. Can you explain me how to use it or provide me some examples? I searched on the internet and I found nothing.
Thank you in advance!

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, gesture callbacks are parameters on node composables:

```kotlin
ModelNode(
    modelInstance = model,
    onTap = { event, node -> /* single tap */ },
    onDoubleTap = { event, node -> /* double tap */ },
    onLongPress = { event, node -> /* long press */ }
)
```

For camera orbit/pan/zoom (move gestures), these are handled by the `cameraManipulator` on `SceneView`. You can customize or disable specific gestures through the manipulator parameters.
